src/docfx/build/xref/XrefResolver.cs

```diff
@@ -47,7 +49,14 @@ internal class XrefResolver
         _redirectionProvider = redirectionProvider;
         _dependencyMapBuilder = dependencyMapBuilder;
         _fileLinkMapBuilder = fileLinkMapBuilder;
+
         _xrefHostName = string.IsNullOrEmpty(config.XrefHostName) ? config.HostName : config.XrefHostName;
+
+        if (_xrefHostName == config.AlternativeHostName)
```

Comment:
I don't think it is necessary to have this validation - Errors.cs is for user error, but here hostname configs/settings are fully controlled by our pipeline. We should and can make sure they are correctly set.